File: caller/src/calls.js
const request = require('request-promise-native');

const webHost = process.env.WEB_HOST || 'http://35.184.201.202';
const webPort = process.env.WEB_PORT || '8080';
const webEndpoint = webHost + ":" + webPort;

module.exports = {
    login: async function () {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/user/login',
                method: 'POST',
                body: {
                    name: 'user',
                    password: 'password'
                },
                json: true
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("login - Error: " + response.statusMessage) :
                        resolve(body);
            });
        });
    },
    getUser: async function () {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/user/uniqueid',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getUser - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getAllCategories: async function () {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/catalogue/categories',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getAllCategories - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getAllProducts: async function () {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/catalogue/products',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getAllProducts - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getProduct: async function (sku) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/catalogue/product/' + sku,
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getProduct - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    rate: async function (sku, mark) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/ratings/api/rate/' + sku + '/' + mark,
                method: 'PUT'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("rate - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getRating: async function (sku) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/ratings/api/fetch/' + sku,
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getRating - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    addToCart: async function (uuid, sku) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/cart/add/' + uuid + '/' + sku + '/1',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("addToCart - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    updateCart: async function (uuid, sku) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/cart/update/' + uuid + '/' + sku + '/2',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("updateCart - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getCart: async function (uuid) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/cart/cart/' + uuid,
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getCart - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getCodes: async function () {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/shipping/codes',
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getCodes - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    getCities: async function (code) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/shipping/cities/' + code,
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("getCities - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    calculateShipping: async function (cityUuid) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/shipping/calc/' + cityUuid,
                method: 'GET'
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("calculateShipping - Error: " + response.statusMessage) :
                        resolve(JSON.parse(body));
            });
        });
    },
    confirmShipping: async function (uuid, shipping) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/shipping/confirm/' + uuid,
                method: 'POST',
                body: shipping,
                json: true
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("confirmShipping - Error: " + response.statusMessage) :
                        resolve(body);
            });
        });
    },
    pay: async function (uuid, cart) {
        return new Promise((resolve, reject) => {
            request({
                url: webEndpoint + '/api/payment/pay/' + uuid,
                method: 'POST',
                body: cart,
                json: true
            }, (error, response, body) => {
                error ?
                    reject(error) :
                    response.statusCode < 200 || response.statusCode > 299 ?
                        reject("pay - Error: " + response.statusMessage) :
                        resolve(body);
            });
        });
    }
};

File: caller-python/calls.py
import requests
import os

timeout = 10
web_host = os.getenv('WEB_HOST', 'http://35.184.201.202:8080')


def login():
    requests.post(web_host + '/api/user/login',
                  data={'name': 'user', 'password': 'password'},
                  timeout=timeout)


def get_user():
    r = requests.get(web_host + '/api/user/uniqueid',
                     timeout=timeout)
    return r.json()


def get_all_categories():
    r = requests.get(web_host + '/api/catalogue/categories',
                     timeout=timeout)
    return r.json()


def get_all_products():
    r = requests.get(web_host + '/api/catalogue/products',
                     timeout=timeout)
    return r.json()


def get_product(sku):
    r = requests.get(web_host + '/api/catalogue/product/' + sku,
                     timeout=timeout)
    return r.json()


def rate_product(sku, mark):
    r = requests.put(web_host +
                     '/api/ratings/api/rate/' + sku +
                     '/' + str(mark),
                     timeout=timeout)
    return r.json()


def get_rating(sku):
    r = requests.get(web_host + '/api/ratings/api/fetch/' + sku,
                     timeout=timeout)
    return r.json()


def add_to_cart(uuid, sku):
    r = requests.get(web_host + '/api/cart/add/' + uuid + '/' + sku + '/1',
                     timeout=timeout)
    return r.json()


def update_cart(uuid, sku):
    r = requests.get(web_host + '/api/cart/update/' + uuid + '/' + sku + '/2',
                     timeout=timeout)
    return r.json()


def get_cart(uuid):
    r = requests.get(web_host + '/api/cart/cart/' + uuid,
                     timeout=timeout)
    return r.json()


def get_codes():
    r = requests.get(web_host + '/api/shipping/codes',
                     timeout=timeout)
    return r.json()


def get_cities(code):
    r = requests.get(web_host + '/api/shipping/cities/' + code,
                     timeout=timeout)
    return r.json()


def calculate_shipping(city_uuid):
    r = requests.get(web_host + '/api/shipping/calc/' + city_uuid,
                     timeout=timeout)
    return r.json()


def confirm_shipping(uuid, shipping):
    r = requests.post(web_host + '/api/shipping/confirm/' + uuid,
                      json=shipping,
                      timeout=timeout)
    return r.json()


def pay_cart(uuid, cart):
    r = requests.post(web_host + '/api/payment/pay/' + uuid,
                      json=cart,
                      timeout=timeout)
    return r.json()
